cub/agent/agent_reduce.cuh

```diff
@@ -377,14 +377,17 @@ struct AgentReduce
     even_share.block_offset += even_share.block_stride;
 
     // Consume subsequent full tiles of input
-    while (even_share.block_offset + TILE_ITEMS <= even_share.block_end)
+    while (even_share.block_offset <= even_share.block_end - TILE_ITEMS)
     {
       ConsumeTile<false>(thread_aggregate,
                          even_share.block_offset,
                          TILE_ITEMS,
                          Int2Type<true>(),
                          can_vectorize);
-      even_share.block_offset += even_share.block_stride;
+      const OffsetT new_offset = even_share.block_offset + even_share.block_stride;
+      even_share.block_offset = (new_offset > even_share.block_offset) 
```

Comment:
Won't the following be better:

```
if (even_share.block_end - even_share.block_offset > even_share.block_stride)
{
    even_share.block_offset += even_share.block_stride;
}
```

Reply:
I believe It would be best to invert the condition
```cpp
if (even_share.block_end - even_share.block_offset < even_share.block_stride)
{
    early return here
}
even_share.block_offset += even_share.block_stride;
...
```

Reply:
@miscco, could you elaborate on why it is better?

Reply:
```cpp
const OffsetT new_offset = even_share.block_offset + even_share.block_stride;
```
Could potentially overflow afaik. Also I would believe, that it is generally always better to early return

Reply:
@miscco the potential overflow is handled below. Early exit adds about 7% to the generated kernel size, so I was suspicious about the suggestion. I've benchmarked the change and there seems to be no huge difference in performance. Since I don't have preferences here, will trust your judgement.

Reply:
Urgh, then please leav this as is. If there is no difference but a regression in kernel size I am all fine with keeping it as is